test/OptionsTest.php

```diff
@@ -161,59 +153,28 @@ public function testPhp2(): void
         $this->assertSame(Options::PHP8, $options->phpVersion(), 'php version is 8.');
     }
 
-    public function testDiagram1(): void
-    {
-        $opt = [
-            'class-diagram' => true,
-        ];
-
-        $options = new Options($opt);
-
-        $this->assertSame(Options::DIAGRAM_CLASS, $options->diagram(), 'diagram is class.');
-    }
-
-    public function testDiagram2(): void
-    {
-        $opt = [
-            'package-diagram' => true,
-        ];
-
-        $options = new Options($opt);
-
-        $this->assertSame(Options::DIAGRAM_PACKAGE, $options->diagram(), 'diagram is package.');
-    }
-
-    public function testDiagram_division(): void
-    {
-        $opt = [
-            'division-diagram' => true,
-        ];
-
-        $options = new Options($opt);
-
-        $this->assertSame(Options::DIAGRAM_DIVISION, $options->diagram(), 'diagram is division.');
-    }
-
-    public function testDiagram3(): void
+    /**
+     * @param string[] $options
+     *
+     * @dataProvider provideDiagrams
+     */
+    public function testDiagrams(array $options, string $expected): void
     {
-        $opt = [];
+        $options = new Options($options);
 
-        $options = new Options($opt);
-
-        $this->assertSame(Options::DIAGRAM_CLASS, $options->diagram(), 'default diagram is class.');
+        $this->assertSame($expected, $options->diagram(), sprintf('diagram is %s.', $expected));
     }
 
-    public function testDiagram4(): void
+    private function provideDiagrams(): array
```

Comment:
In PHP 8.0, references seem to fail if the dataProvider method is `private`.
I would like to see the dataProvider method changed to `public`.

https://www.reddit.com/r/PHP/comments/xupzk8/phpunit_data_provider_methods_can_have/

Reply:
yeah I saw that :D thanks :D